libmultilabel/data_utils.py

```diff
@@ -165,20 +164,23 @@ def load_or_build_text_dict(
     if os.path.exists(embed_file):
         logging.info(f'Load pretrained embedding from file: {embed_file}.')
         embedding_weights = get_embedding_weights_from_file(vocabs, embed_file, silent)
-        vocabs.set_vectors(vocabs.stoi, embedding_weights, dim=embedding_weights.shape[1])
+        dim = torch.as_tensor(embedding_weights).shape[1]
+        vocabs.set_vectors(vocabs.stoi, torch.Tensor(embedding_weights), dim=dim)
     elif not embed_file.isdigit():
         logging.info(f'Load pretrained embedding from torchtext.')
         vocabs.load_vectors(embed_file, cache=embed_cache_dir)
     else:
         raise NotImplementedError
 
     if normalize_embed:
-        embedding_weights = copy.deepcopy(vocabs.vectors.detach().cpu().numpy())
+        # `embedding_weights` is converted to torch.FloatTensor in `vocabs.set_vectors`.
```

Comment:
how about:

vocabs.vectors is a torch.FloatTensor from the result of `vocabs.set_vectors` earlier. To have better precision for calculating the normalization, we use the original `embedding_weights,` a torch.DobleTensor, if it is available.

Reply:
Great a second!